Mathlib/Util/IncludeStr.lean

```diff
@@ -0,0 +1,18 @@
+/-
+Copyright (c) 2021 Henrik Böving. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Henrik Böving, Xubai Wang
+-/
+import Lean
+
+namespace Mathlib.Util
+
+elab (name := includeStr) "include_str " str:str : term => do
+  let some str := str.isStrLit? | Lean.Elab.throwUnsupportedSyntax
+  let srcPath := System.FilePath.mk (← read).fileName
+  let currentDir ← IO.currentDir
```

Comment:
Unused variable.

Reply:
Can this problem solved by adding an `EnvExtension` to lake and call it from `include_str`?

Reply:
This is certainly an interesting idea.  We should try to use relative paths in that extension so that this will work with caching in the future.

The environment extension won't work for everything though.  For example, `example : String := include_str "foo.txt"` would not record the dependency because examples don't modify the environment.

Reply:
> For example, `example : String := include_str "foo.txt"` would not record the dependency because examples don't modify the environment.

Sorry, but I think `include_str` can modify the environment on itself? It is a `TermElab` and have mutable access to `Environment` within `CoreM`'s state.

Reply:
Yes, `include_str` can indeed modify the environment.  But the `example` command then discards all of the changes.